utils_linux.go

```diff
@@ -196,11 +197,20 @@ func (r *runner) run(config *specs.Process) (int, error) {
 		r.destroy()
 		return -1, err
 	}
+
+	if len(r.listenFDs) > 0 && r.preserveFDs > 0 {
+		return -1, fmt.Errorf("Preserve FDs and socket activation (LISTEN_FDS) are mutually incompatible")
+	}
```

Comment:
You an just start by using `len(process.ExtraFiles)`, adding 3 stdio and always preserve from what is already in the `ExtraFiles`